app/scripts/utils/functional.es6

```diff
@@ -81,3 +108,11 @@ Result.mapObj = (fn, obj) => reduceResult(fn, (a, b) => a.concat(b), [], asArray
  * @returns {Result<object>} The merged object in a Result.Ok
  */
 Result.merge = (data) => reduceResult((v) => Ok(v), mergeTwo, {}, data);
+
+/**
+ * @param {array<object>} data Some objects to be aggregated and wrapped in Result.Ok
+ * @param {string} aggProp the property name to aggregate
+ * @param {string} countProp the property name to count
+ * @returns {Result<object>} The object in a Result.Ok
+ */
+Result.aggregate = (data, aggProp, countProp) => reduceResult(checkData, (agg, item) => sumAggregate(agg, item, aggProp, countProp), {keys: {aggProps: [], sumProps: []}}, data);
```

Comment:
I think I get it now:

Given some data like

```js
const data = [
  {a: 'x', b: 'm'},
  {a: 'x', b: 'n'},
  {a: 'y', b: 'n'},
  {a: 'z', b: 'o'},
];
```

`Result.aggregate(data, 'a', 'b');` will make something like

```js
Ok({
  keys: {
    aggProps: ['x', 'y', 'z'],
    sumProps: ['m', 'n', 'o'],
  },
  x: {m: 1, n: 1},
  y: {n: 1},
  z: {o: 1}
});
```

... do I have that right?